manager/apis/app/v1alpha1/plotter_types.go

```diff
@@ -4,21 +4,306 @@
 package v1alpha1
 
 import (
+	"encoding/json"
+	"fmt"
+	"reflect"
+	"strings"
+
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
+// AssetDetails is a list of assets used in the fybrikapplication. In addition to assets declared in
+// fybrikapplication, AssetDetails list also contains assets that are allocated by the control-plane
+// in order to serve fybrikapplication
+type AssetDetails struct {
+
+	// AssetID identifies the asset to be used for accessing the data
+	// +required
+	AssetID string `json:"assetId"`
+
+	// AdvertisedAssetID links this asset to asset from fybrikapplication and is used by user facing services
+	// +optional
+	AdvertisedAssetID string `json:"advertisedAssetId,omitempty"`
+
+	// Cluster name where the asset resides
+	// +required
+	ClusterName string `json:"cluster"`
```

Comment:
We have a DataStore structure that defines the asset metadata and is enough for accessing the asset. 
I don't see a reason for 'ClusterName'.